__tests__/tooling/lib/flatten-schema.test.js

```diff
@@ -186,41 +186,89 @@ describe('$ref usages', () => {
       { name: 'documentReferences[].documentID', type: 'String', description: 'Document ID' },
     ]);
   });
+
+  describe('circular $ref', () => {
+    const oas = {
+      components: {
+        schemas: {
+          Customfields: {
+            type: 'array',
+            items: {
+              $ref: '#/components/schemas/Customfields',
+            },
+          },
+        },
+      },
+    };
+
+    it('should not recurse a circular $ref inside of an objects properties', async () => {
+      const circularSchema = {
+        type: 'array',
+        items: {
+          type: 'object',
+          properties: {
+            id: {
+              type: 'number',
+            },
+            fields: {
+              $ref: '#/components/schemas/Customfields',
+            },
+          },
+        },
+      };
+
+      expect(await flattenSchema(circularSchema, oas)).toStrictEqual([
+        { name: 'id', type: 'Number', description: undefined },
+        { name: 'fields', type: 'Circular', description: undefined },
+      ]);
+    });
+
+    it('should not recurse a circular $ref inside of an array', async () => {
+      const circularSchema = {
```

Comment:
Would we need to test $anyOf, $allOf and such as well?

Reply:
No we shouldn't need to since if a `$ref` is anywhere, and it's circular according to `json-schema-ref-parser` it'll remain as a `$ref: <pointer>` in the schema.